### ISSUE TITLE
routerLink literal escapes semicolon (:) character for Aux routes

### ISSUE THREAD

**Issue by intellix** [angular/angular#12371]
**I'm submitting a ...**  (check one with "x")

```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

Create a link like: `<a routerLink="/home(right:account)">Account</a>` and you will get an output like: `<a href="/home(right%3Aaccount)">Account</a>`

**Expected behavior**

A unescaped href `<a href="/home(right:account)">Account</a>`

**Minimal reproduction of the problem with instructions**

1) Go to: http://plnkr.co/edit/GR1SltnQwwX7Ux37ABf7?p=preview
2) Hover-over/inspect the link to see the generated href

**What is the motivation / use case for changing the behavior?**

routerLink can be used like: `/home`, so I feel like it should also be possible to specify auxiliary routes as well. I wanted to specify this to get around other pending issues.

**Please tell us about your environment:**

**OS**: OSX Sierra
- **Angular version:** 2.0.X
  
  
  
  **Angular**: 2.1.0
- **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
  
  
  
  **Chrome**: Version 54.0.2840.59 (64-bit)
  **Firefox**: 49.0.1
  **Safari**: Version 10.0 (12602.1.50.0.10)
- **Language:** [all | TypeScript X.X | ES6/7 | ES5]
  **TypeScript**: 2.0.2
- **Node (for AoT issues):** `node --version` =

**Comment by DzmitryShylovich** [comment]
wasn't it fixed here https://github.com/angular/angular/commit/602522beb29208626a4593b7cf49d1bf3adbe3a1?

**Comment by polyglotinc** [comment]
I have the same problem but with the hash character!
And this is with Angular 4.3.1

**Comment by intellix** [comment]
I just checked the Plnkr, added the version: 4.3.3 and it doesn't seem to be fixed @DzmitryShylovich

**Comment by jasonaden** [comment]
This should have been fixed with #17890. If the issue persists, please log a new issue.
